File: spring-boot-security-jwt-auth-mongodb-master/src/main/java/com/knf/dev/repository/RoleRepository.java
package com.knf.dev.repository;

import java.util.Optional;

import org.springframework.data.mongodb.repository.MongoRepository;

import com.knf.dev.models.ERole;
import com.knf.dev.models.Role;

public interface RoleRepository extends MongoRepository<Role, String> {

	Optional<Role> findByName(ERole name);
}

File: kotlin-spring-main/kotlin-springsecurity-jwt/src/main/kotlin/com/knf/dev/demo/kotlinspringsecurityjwt/repository/RoleRepository.kt
package com.knf.dev.demo.kotlinspringsecurityjwt.repository

import org.springframework.data.mongodb.repository.MongoRepository
import com.knf.dev.demo.kotlinspringsecurityjwt.model.ERole
import com.knf.dev.demo.kotlinspringsecurityjwt.model.Role
import java.util.*

interface RoleRepository : MongoRepository<Role?, String?> {
    fun findByName(name: ERole?): Optional<Role?>?
}